stonesoup/sensormanager/base.py

```diff
@@ -0,0 +1,147 @@
+                          "cost of making a measurement. The values returned may be scalar or "
+                          "vector in the case of multi-objective optimisation. Metrics may be of "
+                          "any type and in any units.")
+
+    @abstractmethod
+    def choose_actions(self, *args, **kwargs):
+        """A method which returns a set of actions, designed to be enacted by a sensor, or
+        sensors, chosen by some means. This will likely make use of optimisation algorithms.
+
+        Returns
+        -------
+        : dict {:class:`~.Sensor`: [:class:`~.Action`]}
+            Key-value pairs of the form 'sensor: actions'. In the general case a sensor may be
+            given a single action, or a list. The actions themselves are objects which must be
+            interpretable by the sensor to which they are assigned.
+        """
+        raise NotImplementedError
+
+
+class RandomSensorManager(SensorManager):
+    """As the name suggests, a sensor manager which returns a random choice of action or actions
+    from the list available. Its practical purpose is to serve as a baseline to test against.
+
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def choose_actions(self, tracks_list, timestamp, nchoose=1, *args, **kwargs):
+        """Returns a randomly chosen [list of] action(s) from the action set for each sensor.
+
+        Parameters
+        ----------
+        tracks_list: list of :class:`~Track`
+            List of tracks at given time. Used in reward function.
+        timestamp: :class:`datetime.datetime`
+            Time at which the actions are carried out until
+        nchoose : int
+            Number of actions from the set to choose (default is 1)
+
+        Returns
+        -------
+        : dict
+            The pairs of :class:`~.Sensor`: [:class:`~.Action`] selected
+        """
+
+        configs = [dict() for _ in range(nchoose)]
+        for sensor_action_assignment in configs:
+            for sensor in self.sensors:
+                action_generators = sensor.actions(timestamp)
+                chosen_actions = []
+                for action_gen in action_generators:
+                    chosen_actions.append(random.choice(list(action_gen)))
+                sensor_action_assignment[sensor] = chosen_actions
+
+        return configs
+
+
+class BruteForceSensorManager(SensorManager):
+    """A sensor manager which returns a choice of action from those available. The sensor manager
+    iterates through every possible configuration of sensors and actions and
+    selects the configuration which returns the maximum reward as calculated by a reward function.
+
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def choose_actions(self, tracks_list, timestamp, nchoose=1, *args, **kwargs):
```

Comment:
Does this need tracks in a list?

Reply:
No it probably doesn't, would it be better as a set or collection?

Reply:
I think `Set` as that's what commonly used, and just call the variable `tracks`.